src/Crawler.php

```diff
@@ -202,9 +223,16 @@ protected function addAllLinksToCrawlQueue(string $html, Url $foundOnUrl)
                 return $this->crawlQueue->has($url);
             })
             ->each(function (Url $url) use ($foundOnUrl) {
-                $this->crawlQueue->add(
-                    CrawlUrl::create($url, $foundOnUrl)
-                );
+
+				$newNode = null;
```

Comment:
This seems off. Why do you create a null variable? Can this be cleaned up?

Reply:
So this and your other request for not using the reference & were the only way I could figure out how to return the exact node in the tree for the new node. I can definitely look for ways to make this better if you won't accept it otherwise.